### PR TITLE
Support for coordinate as input in the searchbar

### PR DESCRIPTION
**Pull request by pakb**
[Test link](https://web-mapviewer.dev.bgdi.ch/feature_BGDIINF_SB-1438_coordinate_in_searchbar/index.html)

@davidoesch could you give it a look and give me a feedback on how it is?

### CODE REVIEW

**Comment by pakb** [comment]
@davidoesch @boecklic this is now ready, I've fixed the balloon marker placement).
Could you give this a review?

**Comment by davidoesch** [comment]
added 6 tickets in https://jira.swisstopo.ch/browse/BGDIINF_SB-1438

**Comment by pakb** [comment]
@davidoesch I think I'm done with all the feedbacks you've given, thanks!
Could you give it another round of try and tell me if that looks ok now?

**Comment by davidoesch** [comment]
see feedbacks in subtasks of  https://jira.swisstopo.ch/browse/BGDIINF_SB-1438

File: src/api/layers.api.js [hunk 1]
```diff
@@ -190,24 +190,29 @@ const generateClassForLayerConfig = (layerConfig) => {
...
+            reject('API base URL is undefined');
+        } else {
+            const layersConfig = [];
+            axios.get(`${API_BASE_URL}rest/services/all/MapServer/layersConfig?lang=${lang}`)
+                .then(({data: rawLayersConfig}) => {
+                    if (Object.keys(rawLayersConfig).length > 0) {
+                        Object.keys(rawLayersConfig).forEach(rawLayerId => {
+                            const rawLayer = rawLayersConfig[rawLayerId];
+                            const layer = generateClassForLayerConfig(rawLayer);
+                            if (layer) layersConfig.push(layer)
```
Comment (ltshb) [inline src/api/layers.api.js:204]: Code style: I'm not a big fan of inline if ;-)

**Review by ltshb — COMMENTED** [review]
As far as I understand Javascript, looks good, but regarding VueJs and other frameworks no idea ;-), just minor comments.

File: src/utils/coordinateUtils.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+// 47°38'48'' 7°38'48'' or 47°38'48" 7°38'48"
+const REGEX_MERCATOR_WITH_DEGREES_MINUTES = /^\s*([\d]{1,3})[° ]+([\d]{1,2})[' ]+([\d.]+)['"]{0,2}\s*[,/]?\s*([\d]{1,3})[° ]+([\d.]+)[' ]+([\d.]+)['"]{0,2}\s*$/i;
+// 47°38'48''N 7°38'48''E or 47°38'48"N 7°38'48"E
+const REGEX_MERCATOR_WITH_DEGREES_MINUTES_AND_CARDINAL_POINT = /^\s*([\d]{1,3})[° ]+\s*([\d]{1,2})[' ]+\s*([\d.]+)['"]*([NSEW]?)\s*[,/]?\s*([\d]{1,3})[° ]+\s*([\d.]+)[' ]+\s*([\d.]+)['"]*([NSEW]?)\s*$/i;
+
+// LV95, LV03, metric WebMercator (EPSG:3857)
+const REGEX_METRIC_COORDINATES = /^\s*([\d]{1,3}[ ']?[\d]{1,3}[ ']?[\d.]{3,})[\t ,./]+([\d]{1,3}[ ']?[\d]{1,3}[ ']?[\d.]{3,})/i;
+
+// Military Grid Reference System (MGRS)
+const REGEX_MILITARY_GRID = /^3[123][\sa-z]{3}[\s\d]*/i;
```
[the remark this answers is not in the document]
Comment (pakb) [reply src/utils/coordinateUtils.js:20]: yes, in a sense. That would be [this e2e test](https://github.com/geoadmin/web-mapviewer/pull/22/files?file-filters%5B%5D=.js#diff-8fa6803edb863417cc04c42b979016b1bf7d9d10704f103df9f414c0627ee658) and [this unit file](https://github.com/geoadmin/web-mapviewer/pull/22/files?file-filters%5B%5D=.js#diff-32182044f0e0206e48b5f7a896b0cd5d060ab41d5e9d08032ac4e874d7c37e02)

File: src/api/layers.api.js (hunk 1 above)
Comment (pakb) [reply src/api/layers.api.js]: yep, this one slip through, thanks for the spotting

**Review by ltshb — APPROVED** [review]
Although I'm not understanding much of all the code, I'll approve it not to block you ;-)

**Pull request merged**